src/server/node_manager.c

```diff
@@ -8668,22 +8670,14 @@ free_sister_vnodes(job *pjob, char *vnodelist, char *err_msg,
 	}
 	/* increment everything found in new exec_vnode */
 	set_resc_assigned((void *)pjob, 0,  INCR);
-
+						
+	set_scheduler_flag(SCH_SCHEDULE_TERM);
 	rc = send_job_exec_update_to_mom(pjob, err_msg, err_msg_sz,
 							reply_req);
 
 	if (rc == 0) {
-
-		if ((pjob->ji_qs.ji_state == JOB_STATE_RUNNING) &&
-		    (pjob->ji_qs.ji_substate == JOB_SUBSTATE_RUNNING)) {
-			/*
-			 * job must be in a running state
-			 * so as to not conflict with the
```

Comment:
It turned out that I didn't need to enclose the PBS_ACCT_UPDATE and PBS_ACT_NEXT records in an if loop to apply only to running jobs.. I put the if condition here because I was worried that the records would also show up with regular jobs that didn't release any nodes. But that cannot happen since this is the part of the code inside free_sister_nodes() which is only done when pbs_release_nodes or -Wrelease_nodes_on_stageout is called/set.